### PR TITLE
Add back API that some people rely on in torch.cuda.amp.grad_scaler namespace

### PR DESCRIPTION
**Pull request by albanD**

### CODE REVIEW

File: torch/cuda/amp/grad_scaler.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from typing_extensions import deprecated
 
 import torch
+from torch.amp.grad_scaler import OptState
 
 __all__ = ["GradScaler"]
```
Comment (kit1980) [inline torch/cuda/amp/grad_scaler.py:6]: Do we need to add back to `__all__` also?

**Review by kit1980 — APPROVED** [review]

File: torch/cuda/amp/grad_scaler.py (hunk 1 above)
Comment (albanD) [reply torch/cuda/amp/grad_scaler.py:6]: No because it doesn't come from here. So it is not a proper public API (And why I removed it in the first place).

**Review by eqy — APPROVED** [review]

**Comment by kit1980** [comment]
Add `# noqa: F401` to satisfy the linter.

**Pull request closed without merging**